util/Logger.cpp

```diff
@@ -318,6 +325,19 @@ void InitLoggingSystem(const std::string& log_file, const std::string& _unnamed_
     InfoLogger(log) << "Logger initialized at " << std::ctime(&date_time);
 }
 
+void ShutdownLoggingSystemFileSink() {
+    // The file sink may not be safe to use during static deinitialization, because of the
+    // following bug:
+
+    // http://www.boost.org/doc/libs/1_64_0/libs/log/doc/html/log/rationale/why_crash_on_term.html
+    // https://svn.boost.org/trac/boost/ticket/8642
+    // https://svn.boost.org/trac/boost/ticket/9119
+
+    // When either ticket is fixed the ShutdownLoggingSystem() function can be removed.
+
+    GetLoggersToSinkFrontEnds().ShutdownFileSinks();
```

Comment:
Are all file sinks removed, or only those for the relevant process?
e.g. logging of the client stops when the server is stopped